keymap/vim.js

```diff
@@ -247,15 +247,23 @@
     }
 
     function detachVimMap(cm, next) {
-      if (this == CodeMirror.keyMap.vim)
+      if (this == CodeMirror.keyMap.vim) {
         CodeMirror.rmClass(cm.getWrapperElement(), "cm-fat-cursor");
+        if(cm.state.matchBrackets) {
+          cm.state.matchBrackets.fatCursor = false;
```

Comment:
that code is run only once on startup.  This code is run every time we attach/detach the vim keymap (but too early for the startup code).  Per your other comments, I reverted this change.